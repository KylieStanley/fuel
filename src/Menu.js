import React, { Component } from 'react'
import './styles/main.scss';
import { NavLink } from 'react-router-dom'

export class Menu extends Component {
	constructor() {
		super()
		this.state = {
			expanded: false
		}
	}

	toggleMenu = () => {
		this.setState({ expanded: !this.state.expanded })
	}

	render() {
		let toggled = this.state.expanded ? "toggled" : ""

			return (
				<div className={`menu ${toggled}`} onClick={ this.toggleMenu }>
					<div className="hamburger"></div>
				<NavLink to='/main' className="menu-item home">Home</NavLink>
				<NavLink to='/home' className="menu-item faves">Favorites</NavLink>
				<NavLink to='/home' className="menu-item list">Shopping List</NavLink>
				</div>
			)
	}
}

export default Menu