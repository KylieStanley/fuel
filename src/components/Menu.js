import React, { Component } from 'react'
import '../styles/main.scss'
import { NavLink } from 'react-router-dom'

export class Menu extends Component {
	constructor() {
		super()
		this.state = {
			expanded: false
		}
	}

	toggleMenu = () => {
		const expanded = !this.state.expanded
		this.setState({ expanded })
	}

	render() {
		const toggled = this.state.expanded ? 'toggled' : ''

		return (
			<div className={`menu ${toggled}`} onClick={this.toggleMenu}>
				<div className="hamburger" />
				<NavLink to='/main' className="menu-item home">Home</NavLink>
				<NavLink to='/favorites' className="menu-item faves">Favorites</NavLink>
				<NavLink to='/shopping' className="menu-item list">Shopping List</NavLink>
			</div>
		)
	}
}

export default Menu
