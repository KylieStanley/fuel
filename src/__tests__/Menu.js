import React from 'react'
import { shallow } from 'enzyme'
import { Menu } from '../Menu'


describe('Menu', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(
    	<Menu />)
  })

  it('should match the snapshot', () => {
  	expect(wrapper).toMatchSnapshot()
  })

  it('should change the state of expanded if toggled', () => {
  	wrapper.setState({ expanded: false })
  	wrapper.instance().toggleMenu()
    
  	expect(wrapper.instance().state.expanded).toEqual(true)
  })

  it('should have a class of menu toggled if expanded', () => {
  	wrapper.setState({ expanded: false })

  	expect(wrapper.find('div').at(0).hasClass('toggled')).toEqual(false)
  })

  it('should have a class of only menu if closed', () => {
  	wrapper.setState({ expanded: true })

  	expect(wrapper.find('div').at(0).hasClass('toggled')).toEqual(true)
  })

  it('should toggle the state when clicked', () => {
  	const spy = jest.spyOn(wrapper.instance(), 'toggleMenu')
    wrapper.instance().forceUpdate()
    wrapper.find('.menu').simulate('click');

    expect(spy).toHaveBeenCalled();
  })

})
