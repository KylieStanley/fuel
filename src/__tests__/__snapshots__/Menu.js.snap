// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Menu />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="hamburger"
        />,
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="menu-item home"
          to="/main"
        >
          Home
        </NavLink>,
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="menu-item faves"
          to="/home"
        >
          Favorites
        </NavLink>,
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="menu-item list"
          to="/home"
        >
          Shopping List
        </NavLink>,
      ],
      "className": "menu ",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "hamburger",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": "Home",
          "className": "menu-item home",
          "to": "/main",
        },
        "ref": null,
        "rendered": "Home",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": "Favorites",
          "className": "menu-item faves",
          "to": "/home",
        },
        "ref": null,
        "rendered": "Favorites",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": "Shopping List",
          "className": "menu-item list",
          "to": "/home",
        },
        "ref": null,
        "rendered": "Shopping List",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="hamburger"
          />,
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="menu-item home"
            to="/main"
          >
            Home
          </NavLink>,
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="menu-item faves"
            to="/home"
          >
            Favorites
          </NavLink>,
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="menu-item list"
            to="/home"
          >
            Shopping List
          </NavLink>,
        ],
        "className": "menu ",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "hamburger",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "Home",
            "className": "menu-item home",
            "to": "/main",
          },
          "ref": null,
          "rendered": "Home",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "Favorites",
            "className": "menu-item faves",
            "to": "/home",
          },
          "ref": null,
          "rendered": "Favorites",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "Shopping List",
            "className": "menu-item list",
            "to": "/home",
          },
          "ref": null,
          "rendered": "Shopping List",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
